### DISCUSSION TITLE
lost configuration after update

### DISCUSSION THREAD

**Discussion by Chavell3** [arendst/Tasmota#12448] [Q&A] [answered]
Hi,
I had an issue with my flashed tasmota devices.
I inially flashed it to version 6.6.0(i know long time ago) and after some time i wanted to update it to version 6.7.2.
But with the first device I updated, it already lost it's complete configuration and was not accessable through web nor does the buttons had any function.
The default wifi also hasn't been working.
So I'm not able to manage the device completely.
I used the web firmware update function, maybe was that the wrong way, as the had device been reseted before the firmware update was fully done?!

Or is there some special firmware, which I need to use so the configuration will not be reseted after the update?!

I also tried the reset function that should be implemented(reset device 10 time or something) , but that didn't worked either...
And now I "afraid" to try update the other devices too...

Thanks.

**Comment by barbudor** [comment, marked as answer]
The [Upgrading](https://tasmota.github.io/docs/Upgrading/#upgrade-flow) guide recommends to upgrade to [6.7.1](http://ota.tasmota.com/tasmota/release-6.7.1/sonoff.bin), not 6.7.2.
I am unsure if that could be a problem but if you cannot recover access to the device, seriaal-flashing is probably the only solution.

**Reply by Chavell3** [reply]
okay.

But is save to use the tasmota v6.6.0 Web based flash process or should I use flash tools like [](https://github.com/tasmota/tasmotizer) or [](https://github.com/igrr/esptool-ck) instead?
Or if I do the web update, just make sure that my device is also not connected to the same wifi(like the tasmota plug)?
thing is, the equipment for repairing the tasmota device is as ceap/expensive as buying a new one...

**Reply by barbudor** [reply]
Web based flashing, following the documented upgrading path is safe. Use the exact firmwares that are linked in the page (such as the `tasmota-lite.bin`, or after 8.5.1 the `tasmota-lite.bin.gz`
However, the doc list the exact versions to use and if you click on the number you will download the correct version.

There is no reason not to leave the device connected to the same Wifi.

A little investment into some tools like the recommended USD-serial adapters, some wires, a soldering iron is not lost money. With no more than 30€/$30 you can get a correct toolset on AliExpress.

**Reply by Chavell3** [reply]
But, in generell, if upgrading from 6.6.0 to 6.7.1 and later the device should not loose it's configuration, right?

**Reply by barbudor** [reply]
I just flashed sonoff.bin (6.6.0) on a nodemcu, did a few configuration (wifi, mqtt, gpios..)
Then I did a file upload of sonoff-minimal.bin (6.7.1) and finally sonoff.bin (6.7.1) and everything went fine.
I cannot promise that it will go so smoothly in everyone's case but at least it is supposed to do so
